File: ida-chooser.cpp
#include "ida-chooser.h"

using namespace std;


int main(int argc, char** argv)
{

	if (is_first_run())
	{
		string input;
		cout << "Path to IDA folder: ";
		getline(cin, input);
		input = trim(input);
		set_ida_path(input);
	}
	string ida_path = get_ida_path();
	if (get_arch(argv[1]) == 0x8664)
	{
		ida_path = ida_path += "\\ida64.exe\" ";
	}
	else if (get_arch(argv[1]) == 0x14c)
	{
		ida_path = ida_path += "\\ida.exe\"  ";
	}

	system(("\"" + ida_path + argv[1]).c_str());
	return 0;
}

File: src/idaChooser.nim
import os, strutils, osproc
import ./utils

let first_param = paramStr(1)

if isFirstRun():
  print("Path to IDA folder: ")
  var path = readLine(stdin)
  path.stripLineEnd()
  setIDAPath(path)

var ida_path = getIDAPath()
if getFileArch(first_param) == 0x8664:
  ida_path.add(r"\ida64.exe")
else:
  ida_path.add(r"\ida.exe")

discard startProcess(ida_path,args=[first_param])
